library/core/class.cache.php

```diff
@@ -671,6 +671,23 @@ public function makeKey($Key, $Options) {
             $Result = $Prefix.$Key;
         }
 
+        // Make sure key is valid: no control characters or whitespace and no more than 250 characters.
+        if (unicodeRegexSupport()) {
+            // No whitespace or control characters.
+            $replacePattern = '/[\p{Z}\p{C}]/u';
+        } else {
+            // No space, newline, carriage return or tab characters.
+            $replacePattern = '/[\s\n\r\t]/';
```

Comment:
http://php.net/manual/en/regexp.reference.escape.php

\s -> any whitespace characte

The "whitespace" characters are HT (9), LF (10), FF (12), CR (13), and space (32). However, if locale-specific matching is happening, characters with code points in the range 128-255 may also be considered as whitespace characters, for instance, NBSP (A0).